packages/vscode-apollo/src/debug.ts

```diff
@@ -0,0 +1,71 @@
+import { window, workspace, OutputChannel } from "vscode";
+
+// for errors (and other logs in debug mode) we want to print a stack trace showing where they were thrown.
+// This uses an Error's stack trace, removes the three frames regarding this file (since they're useless) and
+// returns the rest of the trace.
+const createAndTrimStackTrace = () => {
```

Comment:
I think it's okay that this logic is duplicated in multiple places (language server and here) since it's only one thing, we don't have a great place to put it, and we could change the formatting based on where it is later.